flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -225,6 +225,17 @@ public static String[] mergeListsToArray(List<String> base, List<String> append)
     //  process parameters
     // ------------------------------------------------------------------------
 
+    public static final ConfigOption<String> FLINK_JAVA_HOME =
+            ConfigOptions.key("env.java.home")
+                    .stringType()
+                    .defaultValue("")
+                    .withDescription(
+                            Description.builder()
+                                    .text(
+                                            "Location where Java is installed. If not specified,"
+                                                    + " Flink will use your default Java installation.")
+                                    .build());
+
```

Comment:
```suggestion
    public static final ConfigOption<String> FLINK_JAVA_HOME =
            ConfigOptions.key("env.java.home")
                    .stringType()
                    .noDefaultValue()
                    .withDescription(
                            Description.builder()
                                    .text(
                                            "Location where Java is installed. If not specified,"
                                                    + " Flink will use your default Java installation.")
                                    .build());

```

changing to noDefaultValue()

Reply:
Thanks, this makes the option semantics clearer.